### DISCUSSION TITLE
Which Version Should I use for development  Elsa 3.0 Pre Release or Elsa 2.2

### DISCUSSION THREAD

**Discussion by mandarshravankar** [elsa-workflows/elsa-core#4195] [Q&A]
Hi. I am working on a .Net project in which I have used Elsa to create workflows and manage activities, I am bit confused between elsa 3.0 and 2.2. which version should I use for development

**Comment by sfmskywalker** [comment]
If you have an existing project that uses Elsa 2, you should keep using that version and not upgrade to Elsa 3.

Elsa 3 is a complete rewrite with no backwards compatibility. This means that if you ever want to move to Elsa 3, you will have to recreate your custom activities and workflows.
